source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -331,22 +336,55 @@ ResponsePtr RawHttpClientImpl::toResponse(Http::ResponseMessagePtr message) {
     return std::make_unique<Response>(errorResponse());
   }
 
+  // Extract headers-to-remove from the storage header coming from the
+  // authorization server.
+  const auto& storage_header_name = Http::Headers::get().EnvoyAuthHeadersToRemove;
+  // If we are going to construct an Ok response we need to save the
+  // headers_to_remove in a variable first.
+  std::vector<Http::LowerCaseString> headers_to_remove;
+  if (status_code == enumToInt(Http::Code::OK)) {
+    // iterate() over the headers rather than just get() in case the storage
+    // header occurs more than once.
+    absl::string_view storage_header_name_view = absl::string_view(storage_header_name.get());
+    message->headers().iterate([&](const Http::HeaderEntry& entry) {
+      if (entry.key() == storage_header_name_view) {
```

Comment:
I'm assuming because we are iterating over the headers `entry` can never be null here.